src/handlers/AnchorHandler.ts

```diff
@@ -0,0 +1,76 @@
+import BigNumber from 'bignumber.js';
+import ContractEntityHandler from './ContractEntityHandler';
+import {
+  AuxiliaryChain,
+  AuxiliaryChainRepository,
+} from '../repositories/AuxiliaryChainRepository';
+import Logger from '../Logger';
+import { AuxiliaryChainRecordNotFoundException } from '../Exception';
+
+/**
+ * This class handles Anchor event
+ */
+export default class AnchorHandler extends ContractEntityHandler<AuxiliaryChain> {
+  private auxiliaryChainRepository: AuxiliaryChainRepository;
+
+  private auxiliaryChainID: number;
+
+  /**
+   * Constructor
+   * @param auxiliaryChainRepository Instance of auxiliary chain repository.
+   * @param auxiliaryChainID Auxiliary chain Id.
+   */
+  public constructor(auxiliaryChainRepository: AuxiliaryChainRepository, auxiliaryChainID: number) {
+    super();
+    this.auxiliaryChainRepository = auxiliaryChainRepository;
+    this.auxiliaryChainID = auxiliaryChainID;
+  }
+
+  /**
+   * This method update latest origin block height.
+   *
+   * @param transactions Bulk transactions.
+   */
+  public async persist(transactions: any[]): Promise<AuxiliaryChain[]> {
+    const chainRecord = await this.auxiliaryChainRepository.get(this.auxiliaryChainID);
+    if (chainRecord == null) {
+      throw new AuxiliaryChainRecordNotFoundException('Cannot find record for auxiliary chain');
+    }
+
+    let anchorBlockHeight = chainRecord.lastOriginBlockHeight;
+    transactions
+      .filter(transaction => chainRecord.anchorAddress === transaction.contractAddress)
```

Comment:
Do we already filter this in graph node (one could specify in the graph node's manifest from which contract address events should arrive)?

Reply:
1. Graph node is an external system. 
2. We have contract Address as a field in thegraph entity.  Multiple anchor contracts events can be written to the same anchor entity.